File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file237.f90
subroutine init_array(n, path)
implicit none

DATA_TYPE, dimension(n,n) :: path
integer :: i, j, n

do i=1, n
do j=1, n
path(j, i) = (DBLE(i * j))/ DBLE(n)
end do
end do
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file237.cpp
void init_array(int n, double path[][100])
{
  int i, j;
  for(i=0; i<n; i++)
  {
    for(j=0; j<n; j++)
    {
      path[j][i] = (double(i * j))/ double(n);
    }
  }
}
